homeassistant/components/mawaqit/mawaqit_wrapper.py

```diff
@@ -0,0 +1,184 @@
+) -> list[dict] | None:
+    """Return mosques in the neighborhood if any. Returns a list of dicts."""
+    nearest_mosques = None
+    try:
+        client = client_instance
+        if client is None:
+            client = AsyncMawaqitClient(
+                latitude, longitude, mosque, username, password, token, session=None
+            )
+        else:
+            # would be better to set pos in client
+            pass
+        await client.get_api_token()
+        nearest_mosques = await client.all_mosques_neighborhood()
+    except BadCredentialsException as e:
+        _LOGGER.error("Error on retrieving mosques: %s", e)
+    except (ConnectionError, TimeoutError) as e:
+        _LOGGER.error("Network-related error: %s", e)
+    finally:
+        if client is not None:
+            await client.close()
+
+    return nearest_mosques
+
+
+async def all_mosques_by_keyword(
+    search_keyword,
+    username: str | None = None,
+    password: str | None = None,
+    token: str | None = None,
+    client_instance: AsyncMawaqitClient | None = None,
+) -> list[dict] | None:
+    """Return mosques in the neighborhood if any. Returns a list of dicts."""
+    search_mosques = []
+    try:
+        client = client_instance
+        if client is None:
+            client = AsyncMawaqitClient(
+                username=username, password=password, token=token, session=None
+            )
+        else:
+            # would be better to set pos in client
+            pass
+        await client.get_api_token()
+
+        if search_keyword is not None:
+            search_mosques = await client.fetch_mosques_by_keyword(search_keyword)
+
+    except BadCredentialsException as e:
+        _LOGGER.error("Error on retrieving mosques: %s", e)
+    except (ConnectionError, TimeoutError) as e:
+        _LOGGER.error("Network-related error: %s", e)
+    finally:
+        if client is not None:
+            await client.close()
+
+    return search_mosques
+
+
+async def fetch_prayer_times(
+    latitude: str | None = None,
+    longitude: str | None = None,
+    mosque: str | None = None,
+    username: str | None = None,
+    password: str | None = None,
+    token: str | None = None,
+    client_instance: AsyncMawaqitClient | None = None,
+) -> dict | None:
+    """Get prayer times from the MAWAQIT API. Returns a dict."""
+    dict_calendar = None
+    try:
+        client = client_instance
+        if client is None:
+            client = AsyncMawaqitClient(
+                latitude, longitude, mosque, username, password, token, session=None
+            )
+        else:
+            # would be better to set pos in client
+            pass
+        await client.get_api_token()
+        dict_calendar = await client.fetch_prayer_times()
+
+    except BadCredentialsException as e:
+        _LOGGER.error("Error on retrieving prayer times: %s", e)
+    except (ConnectionError, TimeoutError) as e:
+        _LOGGER.error("Network-related error: %s", e)
```

Comment:
I think the dataupdatecoordinator base class already logs errors, there's no need to do it here too. Moreover, `BadCredentialsException` should be rethrown as `ConfigEntryAuthFailed`.